property/index.js

```diff
@@ -166,3 +156,68 @@ module.exports = yeoman.generators.Base.extend({
 
   saveProject: actions.saveProject
 });
+
+function coerceDefault(def, value) {
+  var isSupported = typeChoices.indexOf(def.type) !== -1;
+  debug('property type "%s" >> supported: %s', def.type, isSupported);
+
+  if (typeof value === 'string' && isSupported) {
+    switch ((def.type || '').toLowerCase()) {
+      case 'string':
+        if (value === 'uuid' || value === 'guid'){
+          def.defaultFn = value;
+        } else {
+          def.default = value;
+        }
+        break;
+      case 'number':
+        def.default = Number(value);
+        break;
+      case 'boolean':
+        if (['true', '1', 't'].indexOf(value) !== -1 ){
+          def.default = true;
+        } else {
+          def.default = false;
+        }
+        break;
+      case 'object':
+        def.default = JSON.parse(value);
+        break;
+      case 'array':
+        def.default = value.replace(/[\s,]+/g, ',').split(',');
+        break;
+      case 'date':
+        console.warn('Warning: property default value was converted' +
+          ' from string using ISO formatting');
+        if (value.toLowerCase() === 'now'){
+          def.defaultFn = 'now';
+        } else {
+          def.default = new Date(value);
+        }
+        break;
+      case 'geopoint':
+        if (value.indexOf('lat') !== -1 && value.indexOf('lng') !== -1) {
+          def.default = JSON.parse(value);
+        } else {
+          var geo = value.replace(/[\s,]+/g, ',').split(',');
+          def.default = {};
+          def.default.lat = Number(geo[0]);
+          def.default.lng = Number(geo[1]);
+        }
+        break;
+      case 'buffer':
+        console.warn('Warning: property default value was converted' + 
+          'from string using UTF8 encoding');
+        def.default = new Buffer(value);
+        break;
+      case 'any':
+        console.warn('Warning: property default value was stored as string');
+        def.default = value;
+        break;
+      default:
+    }
+    return def;
+  } else {
+    throw Error('Unsupported model property type: ' + def.type);
```

Comment:
> @richardpringle Having the block there can prevent someone from incorrectly mocking the prompt in a test case. 

+1 :heavy_multiplication_x: :100: for throwing a helpful error when the tests is mocking prompts incorrectly.

However, I think that's not the case we have, see this screenshot where I created a property using a custom type "float":

```
$ yo loopback:property
? Select the model: Car
? Enter the property name: mpg
? Property type: (other)
? Enter the type: float
? Required? No

$ cat common/models/car.json
{
  ...
  "properties": {
    "mpg": {
      "type": "float"
    }
  },
  ...
}
```